src/aleph/sdk/utils.py

```diff
@@ -76,3 +77,16 @@ def check_unix_socket_valid(unix_socket_path: str) -> bool:
             unix_socket_path,
         )
     return True
+
+
+def enum_as_str(obj: Union[str, Enum]) -> str:
+    """Returns the value of an Enum, or the string itself when passign a string.
+
+    Python 3.11 adds a new formatting of string enums.
+    `str(MyEnum.value)` becomes `MyEnum.value` instead of `value`.
+    """
+    if isinstance(obj, str):
+        return obj
+    else:
+        assert isinstance(obj, Enum)
+        return obj.value
```

Comment:
This implementation will not work because you check inheritance from `str` before `Enum`.

```
class C(str, Enum):
  V = "v"

c = C.V
isinstance(c, str) # True
```

```suggestion
def enum_as_str(obj: Union[str, Enum]) -> str:
    """Returns the value of an Enum, or the string itself when passign a string.

    Python 3.11 adds a new formatting of string enums.
    `str(MyEnum.value)` becomes `MyEnum.value` instead of `value`.
    """
    if not isinstance(obj, str):
        raise TypeError(f"Unsupported enum type: {type(obj)}")
    
    if isinstance(obj, Enum):
        return obj.value

    return obj
```